### PR TITLE
LH-67707/todo-fixes

### PR DESCRIPTION
**Pull request by weilueluo**
https://jira-eng-rtp3.cisco.com/jira/browse/LH-67707 (or Github issue ID)

### Description
This PR fixes `TODO`s in the codebase, also some quality of life improvements.
- Rename some `ipv4` to `socket_address`.
- Add lint instruction in README.
- Reword some comments.
- Remove unnecessary `fmt.Sprintf`.
- Added a `state` package for storing state machine states.

### CODE REVIEW

File: client/device/delete.go [hunk 1]
```diff
@@ -30,7 +29,6 @@ func NewDeleteRequest(ctx context.Context, client http.Client, deleteInp DeleteI
 	return req
 }
 
-// TODO: Change so func only returns error
```
Comment (weilueluo) [inline client/device/delete.go:33]: kept for consistency

File: provider/internal/device/asa/resource.go [hunk 2]
```diff
@@ -138,9 +138,7 @@ func (r *AsaDeviceResource) Configure(ctx context.Context, req resource.Configur
 	r.client = client
 }
 
-// TODO PlanModifiers for when the user does not enter port in socket_address
 // TODO plan diffing should exclude host, id, port, and sdc_name (unless SDCType is changed, in which case it should be a destroy and create)
-// TODO terraform should error if the credentials entered are incorrect
```
Comment (weilueluo) [inline provider/internal/device/asa/resource.go:143]: removed as these already implemented

**Review by hazital — APPROVED** [review]
LGTM, but upon running the acceptance locally I noticed some failing tests. is it related?

**Comment by weilueluo** [comment]
> LGTM, but upon running the acceptance locally I noticed some failing tests. is it related?

hmm weird, I did not run it locally, but since it passed here, I assume it is just some environment difference that have caused it.

**Pull request merged**